retrace/glretrace.py

```diff
@@ -171,9 +171,38 @@ def retraceFunctionBody(self, function):
             print(r'    if (currentContext && currentContext->features().pixel_buffer_object) {')
             print(r'        glGetIntegerv(GL_PIXEL_PACK_BUFFER_BINDING, &_pack_buffer);')
             print(r'    }')
-            print(r'    if (!_pack_buffer) {')
+            print(r'    if (!_pack_buffer && !retrace::runReadback) {')
             print(r'        return;')
             print(r'    }')
+            print(r'    if (retrace::runReadback) {')
+            if function.name == "glGetTexImage":
+               print(r'     GLint max_tex_size;')
+               print(r'     glGetIntegerv(GL_MAX_TEXTURE_SIZE, &max_tex_size);')
+               print(r'     std::vector<char> buffer(max_tex_size * max_tex_size * max_tex_size);');
+               print(r'     glGetTexImage(call.arg(0).toUInt(), call.arg(1).toSInt(), call.arg(2).toUInt(), call.arg(3).toUInt(), &buffer[0]);');
```

Comment:
Instead of explicitly call `glGetXXX`, I'd simply allocate the output buffer, and let the automatically generated code to call `gl*` follow through.

That is, something like this:

```python
    print(r'    std::vector<char> buffer;')
    print(r'     if (!_pack_buffer) {')
    if function.name == "glGetTexImage":
        print(r'    buffer.resize( FIXME )')
        print(r'    pixels = &buffer[0]')
    elif function.name == "....":
        ...
    else:
        # Don't know how to handle this one, better return
        print(r'   return;')
    ....
    print(r'}')
    # The actual call will be generated later
```
But this needs to happen not here, but rather right before the call.